### ISSUE TITLE
Consistent Crashes 1.19.2

### ISSUE THREAD

**Issue by Poke097** [Tiviacz1337/Travelers-Backpack#558]
**Describe the bug**
I have Feature Placement/Random Access Thread crash when going to specific areas, it seems to include stack traces with Create as well. Disabling the mod seems to fix the issue, it may be a random error or incompatibilities with my mod set.

This occurred on seed -4647545476531700989, at coordinates -1870.74, y=94.82, z=-972.50

https://mclo.gs/sA8UwMb

Here are the logs above,

**To Reproduce**
1. Make a world
2. Explore the world, preferably go near structures.
3. Game will freeze and crash.

**Expected behavior**
I did not expect consistent crashes when going to specific areas or just moving.

**Required information's (please complete the following information):**
- Mod version: travelers-backpack-1.19.2-8.2.14
- Mod Loader: Fabric 0.14.17

**Comment by Zarepheth** [comment]
I've just encountered a similar error. I'm not sure if it has the same cause, but the stack trace starts the same before diverging.

I've also reported this with [Porting-Lib](https://github.com/Fabricators-of-Create/Porting-Lib/issues/41), as it is the other mod in the stack trace.

> Was exploring the Nether when my world suddenly crashed. The stack trace implicates "Porting-Lib" and "Travelers Backpack".
> 
> I'm using the Medieval MC (FABRIC) modpack v18.5, with extra mods and mods I've updated myself instead of waiting on the modpack's author.
> 
> Crash Report: [crash-2023-03-04_20.54.56-server.txt](https://github.com/Fabricators-of-Create/Porting-Lib/files/10890678/crash-2023-03-04_20.54.56-server.txt)

**Comment by Zarepheth** [comment]
In my world the crash occurs near coordinates -259, 59, 161 (x,yz) in the Nether. The game froze after reloading the game from its previous save and wandering towards the same area as the crash report in my previous post.

World Seed: -8479227576142809859

2nd Crash Report (report generated, but game froze and stopped responding, so had to force-close from Windows): [crash-2023-03-04_21.38.30-server.txt](https://github.com/Tiviacz1337/Travelers-Backpack/files/10890734/crash-2023-03-04_21.38.30-server.txt)

**Comment by Zarepheth** [comment]
Additional crash, after disabling Create mod. Now the Porting-Lib mod is coming from Twilight Forest, so I guess multiple mods in the modpack may have embedded it.

**Comment by Poke097** [comment]
> Additional crash, after disabling Create mod. Now the Porting-Lib mod is coming from Twilight Forest, so I guess multiple mods in the modpack may have embedded it.

Hi! May I know what launcher you were using and what extra mods you have? this would help a bunch, also what mods you updated in specific if possible.

And do you by chance have discord and have reported this in Luna Pixel's discord? I saw that you were also using Medieval MC.

Last thing to note, did you try reproducing this without the extra mods on the current pack or just the updated mods?

**Comment by Zarepheth** [comment]
I'm using the CurseForge Overwolf client, which launches the Minecraft launcher with a profile for the modpack I'm playing. (_I presume it's the native Minecraft launcher._)

It will take a while to get the list of extra and updated mods. I really wish CurseForge would make it easy to get the list of mods that differs from the mod author's list. But I do have some tools to help.

I've not yet reported it to Luna Pixel - I wasn't sure whether the "Create" mod as there due to me installing it or if it came with the modpack. However, Twilight Forest does come with the modpack and when I'm done with my work above, I'll also know about Create.

I've setup a new CurseForge profile with the unedited modpack. I'll export my current world settings and import them into a new world with the plain modpack. Then attempt to reproduce the crash. It may take several hours before I report back.

**Comment by Zarepheth** [comment]
While putting together the list of mod differences, I see that Traveler's Backpack is one that I updated.

I also see that Create and Create Addons are not part of the modpack. I don't recall adding them, but it I clicked the wrong place I may have done so. Or perhaps they were previously in the modpack and due to manual changes, when I updated the modpack they remained behind instead of getting removed.

An initial test of the plain modpack does not replace the error. But I was in creative mod at the time, and had no backpack while flying towards the area where the game was crashing. I'm going to test a few things so see what can reproduce the crash.

Mod Differences

Different but with same file name:
- Marium's Soulslike Weaponry v0.9.8 - mariumbacchus

Mods I've added, deliberately or accidentally:
- Animal Feeding Trough v1.03 - Slexom
- AppleSkin v2.4.1 - squeek502
- Aquatic Torches v1 - realmayus
- Arcanus: Legacy v1.3 - Camellias
- Base Origins Tweaks v1 - lankasteria
- Better Enchanted Books v1.4.1 - Bernasss12
- Better Safe Bed v4 - frankv_
- Better Than Mending v1.3.0 - legobmw99
- Better Third Person v1.9.0 - Socolio
- Chalk v2.1.0 - DaFuqsy
- Chalk Colorful Addon v2.11 - DaFuqsy
- Colorful Subtitles v1 - haykam
- Configured v 2.0.0 - MyCrayfish (hmm... did I add this, or did it come as a dependency?)
- Crafting Tweaks v15.1.6 - BlayTheNingth
- Create Crafts & Additions v20230211a - MRHminer (currently disabled, I guess I added it though I don't remember doing s
- Create v0.5.0.i-961 - tropheusj (currently disabled, I guess I added it though I don't remember doing so)
- CreativeCore v2.9.3 - CreativeMD
- Crying Ghasts v 3.1 - Serilum
- Diagonal Fences v 4.2.3 - Fuzs_
- Environmental Creepers v1.6.0 - masady
- Ftb Ultimine v1902.3.5-build.65 - FTB
- Goblin Traders v1.8.1 - MrCrayfish (Hmm, I also see the one from "jab125twitchy" - do they conflict?)
- Grind Enchantments v2.0.0 - mschae23
- Immersive Portals v2.3.1 - qouteall (currently disabled due to conflicts with Sodium or mods that depend upon Sodium)
- ItemPhysic Full v1.6.6 - CreativeMD
- Leaves Us In Peace v1.0.12 - supersaiyansubtlety
- Light Overlay v7.0.0 - shedaniel
- Mob Catcher v1.2.15 - kwpugh
- Overpowered Mending v2.8.2 - rakambda
- Passive Shield v3.2 - Serilum
- Path under fence gates v1.3.0 - Steveplays28
- Precise Block Placing v2.0.0 - Ellemss
- Third Person Maps v1.4.0 - Pepperoni__Jabroni__
- TorchBowMod v1.0 - noriokun4649
- WMITAF v3.1.1 - ShaksterNano

Modpack mods I've disabled:
- EnvironmentZ v1.3.5 - Globox_Z
- Vein Mining v1.0.2 - TheIllusiveC4

Modpack mods I've updated:
- Alternate Current v1.5 (original v1.4) - SpaceWalkerRS
- Balm v4.5.6 (original v4.5.2) - BlayTheNingth
- Better Advancements v0.2.2.147 (original v0.2.2.142) - way2muchnoise
- Better Combat v1.6.1 (original v1.5.5) - daedelus_dev
- Chat Heads v0.10.5 (original v0.9.0) - dzwdz
- Custom Entity Models v0.7.3 (original v0.7.1) - dorianpb
- Better Statistics Screen v2.3 (original v2.2.1) - TheCSDev
- ChoiceTheorem's Overhauld Village v3.1.6 (original v3.1.4) - ChoiceTheorem
- Collective v6.49 (original 6.43) - Serilum
- Combat Roll v1.1.4 (original v1.1.2) - daedelus_dev
- CorgiLib v1.0.0.32 (original v1.0.0.22) - Corgi_Taco
- Dramatic Doors v4.2.3 (original v1.4.1) - FizzWare
- Drippy Loading Screen v1.4.2_1 (original 1.4.1) - Keksuccino
- Emotecraft v2.2.7-b.build.47 (original v2.2.5) - KosmX
- End Goblin Traders v1.5.2 (original v1.5.1) - jab125twitchy
- Entity Culling v1.6.1 (original v1.5.2) - tr9zw
- Entity Texture Features v4.3.2 (original v4.3.1) - traben_0
- Expanded Delight v0.2.5.2 (original v0.2.5.1) - ianm1647
- Fabric API v0.75.1 (original v0.73.2) - modmuss50
- Forge Config API Port v4.2.11 (original v.2.10) - Fuzs_
- FTB Library v1902.3.14-build.184 (original v1902.3.14-build.175) - FTB
- FTB Quests v1902.4.9-build.195 (original v1902.4.6-build.176) - FTB
- Galosphere v1.2.2 (original v1.2.1) - orcinus73
- GeckoLib v3.1.40 (original v3.1.39) - ThanosGecko
- Geophilic - Biome Overhauls v1.2.1 (original v1.0.1) - bebebea_loste
- ImmediatelyFast v1.1.9 (original v1.1.8) - RaphiMC
- Immersive Weathering v1.2.8 (original v1.2.7) - OrdanaryMods
- Kiwi v8.3.0 (original v8.1.2) - Snownee_
- MC Dungeons Armors v2.4.4 (original v2.4.2) - chronos_sacaria
- MC Dungeons Artifacts v1.5.3 (original v1.5.2) - chronos_sacaria
- MC Dungeons Weapons v5.0.5 (original v5.0.4) - chronos_sacaria
- MC Story Mode Armors v3.1.3 (original v3.1.2) - chronos_sacaria
- MedievalWeapons v1.4.3 (original v1.4.2) - Globox_Z
- MemoryLeakFix 1.19.3-0.7.0 (original 1.19.1-0.7.0) - FX_PROCESS
- Mine Cells - Dead Cells Mod v1.4.0 (original v1.3.5) - mim1q
- MmmMmmMmmMmm (Target Dummy) v1.7.1 (original v1.7.0) - MehVahdJukaar
- Mod Menu v4.2.0-beta.2 (original v4.1.2) - ProspectorDev
- Moonlight Lib v2.2.13 (original 2.2.5) - MehVahdJukaar
- Mythic Metals v0.17.2 (original v0.17.0) - Noaaan
- Mythic Upgrades v1.5.1 (original v1.5) - trique06
- NetherPortalFix v10.0.1 (original v10.0.0) - BlayTheNingth
- Night Config Fixes v4.0.5 (original v4.0.2) _ Fuzs_
- Paradise Lost v2.1.0-beta (original 2.0.2-beta) - ImmortalDevs
- Pehkui v3.7.0 (original v3.6.3) - Virtuoel
- Philip's Ruins v5.0 (original v4.6) - philipmoddev
- Pickup Notifier v4.2.4 (original v4.2.0) - Fuzs_
- PlayerAnimator v1.0.2 (original v0.4.2) - KosmX
- Public Gui Announcement v4.3.3 (original v4.3.2) - AbsolumJackdaw
- Puzzle v1.4.2 (original v1.4.0) - Motschen
- Red's More Structures v1.0.6 (original v1.0.4) - redstone3game
- RelicEx v3.3.0 (original v3.2.7) - clevernucleus
- Repuposed Structures v6.3.16 (original v6.1.6) - telepathicgrunt
- Roughly Enough Items v9.1.591 (original v9.1.587) - shedaniel
- Shroom Dealers! v1.6 (original v1.5.2) - Marwinekk256
- Snowy Spirit v2.1.7 (original v2.1.4) - MehVahdJukaar
- Spellblades and Such v1.0.14.0.hotfix 1 (original v1.0.11.1) - cleannrooster
- Spell Engine v0.9.14 (original v0.9.10) - daedelus_dev
- Spell Power v0.9.7 (original v0.9.6) - daedelus_dev
- Starter Kit v5.2 (original v5.0) - Serilum
- Supplementaries v5.2 (original v5.0) - MehVahdJukaar
- TerraBlender v2.0.1.136 (original v2.0.1.130) - TheAdubbz
- The Bumblezone v6.6.16 (original v6.6.7) - telepathicgrunt
- Tom's Simple Storage Mod v1.5.1 (original v1.4.6) - tom54541
- Travelers Backpack v8.2.14 (original v8.2.13) - Tiviacz1337
- HT's TreeChop v0.17.6 (original v0.17.3) - hammertater
- Unsafe World Random Access Detector v1.1.0 (original v1.0.0) - ishlandmc
- Villager Names v4.5.0 (original v4.2.1) - YumatanGames
- Visual Workbench v4.2.3 (original v4.2.2) - Fuzs_
- Visuality v0.5.6 (original v0.5.5) - PinkGoosik
- Wizards v0.9.11 (original v0.9.10) - daedelus_dev
- Alloy Forgery v2.0.19 (original v2.0.17) - gliscowo
- YUNG's API v3.8.8 (original v3.8.2) - YUNGNICKYOUNG
- YUNG's Better Desert Temple v2.2.2 (original v2.2.1) - YUNGNICKYOUNG

**Comment by Poke097** [comment]
> So, testing with the original Medieval MV [FABRIC] v18.5 modpack and updating only Traveler's Backpack (from v8.2.13 to v8.2.14), I get a similar error upon entering the Nether.
> 
> I used the same world for my testing, occasionally renaming the world's "DIM-1" folder to force new Nether generation, as the previous errors seemed related to generating chunks.
> 
> Crash Report: [crash-2023-03-05_21.53.49-server.txt](https://github.com/Tiviacz1337/Travelers-Backpack/files/10894252/crash-2023-03-05_21.53.49-server.txt)
> 
> Using the original modpack with no updates, I was unable to reproduce the error, even on a fresh regeneration of the Nether, even wearing a traveler's backpack. But once I updated just this mod, the crash occurred while rendering the nether after I stepped of the portal. I'm not sure I even moved a whole block, but some of the landscape in my view had rendered before the game froze.

I'm guessing since you had just updated the mod, the world was already generated, meaning whenever you updated it, and went back, the world was trying to load but due to the error most likely in the updated version of the mod, caused the crash when trying to load the world up

**Comment by Zarepheth** [comment]
I downgraded Traveler's Backpack in my main world and went back to the point of the crash. This time, no crash.

I've got no idea how or what in that minor upgrade would cause the error, but downgrading in my 360+ mod main profile prevents the error, while upgrading it in the plain modpack can cause the error.

**Comment by Poke097** [comment]
> I downgraded Traveler's Backpack in my main world and went back to the point of the crash. This time, no crash.
> 
> I've got no idea how or what in that minor upgrade would cause the error, but downgrading in my 360+ mod main profile prevents the error, while upgrading it in the plain modpack can cause the error.

No clue either! must have been something with the update, but thankfully due to your efforts I've been able to let Luna Pixel and other mod devs know, and I believe the dev of this mod has confirmed they are working on it so thank you very much for your assistance!

**Comment by Zarepheth** [comment]
There was a generated structure near the point of my main profile's crash that contains a respawn anchor. I've never before come across such. However, the nearer side of the structure was rendering before the crash. I don't know if that is related, nor what mod (if any) caused the structure generation. The structure and respawn anchor are present in both the plain profile and my main profile.

**Comment by Zarepheth** [comment]
I'm still exploring, in my main profile. I see something that looks like a structure and plan to investigate. I believe it is closer to the chunk generating edge of my original crash location - since that respawn anchor was probably present before the crash.

**Comment by Zarepheth** [comment]
I got close enough to see that it's just a second nether fortress. I wouldn't expect anything special from it.

**Comment by Tiviacz1337** [comment]
@Zarepheth Is it possible for you to contact me on discord to check updated mod file if issue still persists?

**Comment by Poke097** [comment]
> @Zarepheth Is it possible for you to contact me on discord to check updated mod file if issue still persists?

Is this a file or is this released?

**Comment by Poke097** [comment]
I can also test if needed

**Comment by Tiviacz1337** [comment]
@Poke097 If you know exact steps when issue occurs with latest version, add me on discord. I'll send updated mod file later today, so you could test if issue is still present Tiviacz1337#1878

**Comment by Poke097** [comment]
> @Poke097 If you know exact steps when issue occurs with latest version, add me on discord. I'll send updated mod file later today, so you could test if issue is still present Tiviacz1337#1878

Yep added, I believe it's just Poke#0007, I typed in your #bug-report channel a few days ago haha

**Comment by Zarepheth** [comment]
It may be a while before I test it, but thanks!

**Comment by Zarepheth** [comment]
As I mentioned on Discord, @Poke097 sent me the new `*.jar` file. Thus far in my testing and playing it has not been responsible for any crashes.

**Comment by Tiviacz1337** [comment]
https://www.curseforge.com/minecraft/mc-mods/travelers-backpack-fabric/files/all
